### PR TITLE
Add vlan and mtu parameters to OVNOverlayNetworkAttachmentDefinition

### PR DESCRIPTION
**Pull request by Anatw**
##### Short description:
Add vlan and mtu parameters to OVNOverlayNetworkAttachmentDefinition.

##### What this PR does / why we need it:
Needed for the localnet feature.

### CODE REVIEW

File: ocp_resources/network_attachment_definition.py [hunk 1]
```diff
@@ -221,6 +223,8 @@ def __init__(
         )
         self.network_name = network_name
         self.topology = topology
+        self.vlan = vlan
```
Comment (myakove) [inline ocp_resources/network_attachment_definition.py:226]: Missing in docstring

File: ocp_resources/network_attachment_definition.py [hunk 2]
```diff
@@ -230,7 +234,11 @@ def to_dict(self):
             if not self.topology:
                 raise ValueError("topology is required")
             spec_config = self.res["spec"]["config"]
-            spec_config["name"] = self.network_name
```
Comment (myakove) [inline ocp_resources/network_attachment_definition.py:233]: `self.network_name` is required but not used anymore?

**Review by myakove — CHANGES_REQUESTED** [review]

File: ocp_resources/network_attachment_definition.py (hunk 2 above)
Comment (Anatw) [reply ocp_resources/network_attachment_definition.py:233]: It was so important I take care of it in the code and don't trust the user, so this is redundant :)

File: ocp_resources/network_attachment_definition.py (hunk 1 above)
Comment (Anatw) [reply ocp_resources/network_attachment_definition.py:226]: Ack.

File: ocp_resources/network_attachment_definition.py (hunk 2 above)
Comment (myakove) [reply ocp_resources/network_attachment_definition.py:233]: I don't understand this answer.

Comment (Anatw) [reply ocp_resources/network_attachment_definition.py:233]: I removed it entirely. It is taken care of here:
spec_config["netAttachDefName"] = f"{self.namespace}/{self.name}"

File: ocp_resources/network_attachment_definition.py [hunk 3]
```diff
@@ -211,26 +213,33 @@ def __init__(
         https://docs.openshift.com/container-platform/4.14/networking/multiple_networks/configuring-additional-network.html#configuration-ovnk-additional-networks_configuring-additional-network
 
         Args:
-            network_name (str): The name of the network. This field is mandatory when not using yaml
+            network_name (str): The name of the network.
                 file.
             topology (str): The secondary network topology to be created.
+            vlan (int): A vlan tag ID.
```
Comment (myakove) [inline ocp_resources/network_attachment_definition.py:219]: docstring is missing args.

File: ocp_resources/network_attachment_definition.py [hunk 4]
```diff
@@ -211,26 +213,33 @@ def __init__(
         https://docs.openshift.com/container-platform/4.14/networking/multiple_networks/configuring-additional-network.html#configuration-ovnk-additional-networks_configuring-additional-network
 
         Args:
-            network_name (str): The name of the network. This field is mandatory when not using yaml
+            network_name (str): The name of the network.
```
Comment (myakove) [inline ocp_resources/network_attachment_definition.py:216]: This should be name of the network or `namespace/network-name`?

Comment (myakove) [reply ocp_resources/network_attachment_definition.py:216]: add that if not sent, the name will be `{self.namespace}/{self.name}`

**Review by myakove — CHANGES_REQUESTED** [review]

File: ocp_resources/network_attachment_definition.py (hunk 4 above)
Comment (Anatw) [reply ocp_resources/network_attachment_definition.py:216]: Yes, the name of the network should be defined exactly as you wrote - namespace/nad.name

File: ocp_resources/network_attachment_definition.py (hunk 3 above)
Comment (Anatw) [reply ocp_resources/network_attachment_definition.py:219]: Ack.

File: ocp_resources/network_attachment_definition.py [hunk 5]
```diff
@@ -211,26 +213,35 @@ def __init__(
...
-            spec_config["name"] = self.network_name
+            if self.vlan:
+                spec_config["vlanID"] = self.vlan
+            if self.mtu:
+                spec_config["mtu"] = self.mtu
+            spec_config["name"] = self.name
             spec_config["topology"] = self.topology
-            spec_config["netAttachDefName"] = f"{self.namespace}/{self.name}"
+            spec_config["netAttachDefName"] = (
+                self.network_name if self.network_name else f"{self.namespace}/{self.name}"
```
Comment (myakove) [inline ocp_resources/network_attachment_definition.py:245]: Use `or`

**Review by myakove — CHANGES_REQUESTED** [review]

File: ocp_resources/network_attachment_definition.py [hunk 6]
```diff
@@ -211,26 +213,33 @@ def __init__(
         https://docs.openshift.com/container-platform/4.14/networking/multiple_networks/configuring-additional-network.html#configuration-ovnk-additional-networks_configuring-additional-network
 
         Args:
-            network_name (str): The name of the network. This field is mandatory when not using yaml
-                file.
+            network_name (str): The name of the network, including the namespace
+                where the network it available, e.g. namespace/network-name.
             topology (str): The secondary network topology to be created.
+            vlan (int): A vlan tag ID that will be assigned to traffic from this
```
Comment (rnetser) [inline ocp_resources/network_attachment_definition.py:219]:
```suggestion
            vlan (int, optional): A vlan tag ID that will be assigned to traffic from this
```

**Review by rnetser — CHANGES_REQUESTED** [review]

File: ocp_resources/network_attachment_definition.py (hunk 6 above)
Comment (rnetser) [reply ocp_resources/network_attachment_definition.py:219]: not addressed

**Review by rnetser — CHANGES_REQUESTED** [review]

File: ocp_resources/network_attachment_definition.py (hunk 6 above)
Comment (Anatw) [reply ocp_resources/network_attachment_definition.py:219]: Ack

File: ocp_resources/network_attachment_definition.py [hunk 7]
```diff
@@ -211,26 +213,33 @@ def __init__(
...
+                additional network.
+            mtu (str): The maximum transmission unit (MTU).
```
[the remark this answers is not in the document]
Comment (Anatw) [reply ocp_resources/network_attachment_definition.py:221]: Ack

File: ocp_resources/network_attachment_definition.py [hunk 8]
```diff
@@ -211,26 +213,33 @@ def __init__(
         https://docs.openshift.com/container-platform/4.14/networking/multiple_networks/configuring-additional-network.html#configuration-ovnk-additional-networks_configuring-additional-network
 
         Args:
-            network_name (str): The name of the network. This field is mandatory when not using yaml
-                file.
+            network_name (str): The name of the network, including the namespace
```
Comment (rnetser) [inline ocp_resources/network_attachment_definition.py:216]:
```suggestion
            network_name (str, optional): The name of the network, including the namespace
```

**Review by rnetser — CHANGES_REQUESTED** [review]

File: ocp_resources/network_attachment_definition.py (hunk 8 above)
Comment (Anatw) [reply ocp_resources/network_attachment_definition.py:216]: Ack

**Review by rnetser — APPROVED** [review]

**Review by myakove — APPROVED** [review]

**Comment by NedAnonymous7509** [comment]
What does it do???

**Pull request merged**